### PR TITLE
fix: massage commit messages

### LINKED ISSUE #12062: Commit messages should not be fully lowercased
### How are you running Renovate?

WhiteSource Renovate hosted app on github.com

### Please select which platform you are using if self-hosting.

_No response_

### If you're self-hosting Renovate, tell us what version of Renovate you run.

_No response_

### Describe the bug

When Renovate opens a PR with semantic commits the entire commit message/PR title is lowercased. This is unnecessary, only the first letter needs to be lowercased ([per Angular commit guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#summary), I think most other semantic commit specs do the same).

Given the title:

```
fix(deps): Update My Org dependencies
```

Actual behavior:

```
fix(deps): update my org dependencies
```

Expected:

```
fix(deps): update My Org dependencies
```

Relevant code:
https://github.com/renovatebot/renovate/blob/00e191795ce9a56752ece1891b41d0b6b92cba77/lib/workers/repository/updates/generate.ts#L184-L189

### Relevant debug logs

_No response_

### Have you created a minimal reproduction repository?

No reproduction repository

### PR DESCRIPTION
**Pull request by RahulGautamSingh**
## Changes:

Updated commit-message builder to only lowercase first word of commit messages.

## Context:

Fixes #12062 

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/workers/repository/model/commit-message.ts [hunk 1]
```diff
@@ -44,6 +46,15 @@ export class CommitMessage {
     return [prefix, message].join(' ').trim();
   }
 
+  public static formatCasing(commitMessage: string): string {
```
Comment (viceice) [inline lib/workers/repository/model/commit-message.ts:49]: I still think we should do it before we generate the full message.

Comment (RahulGautamSingh) [reply lib/workers/repository/model/commit-message.ts:49]:
```js
if (upgrade.semanticCommitCasing) {
      // We only need to lowercase the first word of the commitMessage
      const splitMessage = upgrade.commitMessage.split('\n');
      splitMessage[0] = CommitMessage.formatCasing(splitMessage[0]);
      upgrade.commitMessage = splitMessage.join('\n');
    }
    ```
We only lowercase the first word incase `upgrade.semanticCommitCasing` is true, so is it necessary to change the before this.

Comment (rarkins) [reply lib/workers/repository/model/commit-message.ts:49]: Can't it be done somewhere around https://github.com/renovatebot/renovate/blob/0e2f83ae6d7a5297f9ee4d9990245b063e3942ae/lib/workers/repository/model/commit-message.ts#L44 and avoid the regex?

Comment (RahulGautamSingh) [reply lib/workers/repository/model/commit-message.ts:49]: We have 4 different types of commit-messages so using regex is the better approach I think.

We can put the functionality in `toString` method and use it in the `generate.ts` file. There is no problem I could find  with that except the case that it has been used in two other files and I am not sure if we want to change those commit-messages or not.
The 2 files are:-
https://github.com/renovatebot/renovate/blob/8c588ddbfc4a6deba1521401573422485163db67/lib/workers/repository/onboarding/branch/create.ts#L41

https://github.com/renovatebot/renovate/blob/8c588ddbfc4a6deba1521401573422485163db67/lib/workers/repository/onboarding/branch/rebase.ts#L58

Comment (RahulGautamSingh) [reply lib/workers/repository/model/commit-message.ts:49]: @RahulGautamSingh @viceice Could you please read the above comment, and guide as to what changes need to be made.

Comment (rarkins) [reply lib/workers/repository/model/commit-message.ts:49]: We don't want to use regex for this. Can you find a way?

Comment (RahulGautamSingh) [reply lib/workers/repository/model/commit-message.ts:49]: On it.

File: lib/workers/repository/model/commit-message.ts [hunk 2]
```diff
@@ -44,6 +44,27 @@ export class CommitMessage {
...
+  static lowerCaseFirstWord(message: string): string {
+    let copyMsg: any = message;
+    if (!copyMsg) {
+      return copyMsg;
+    }
+    copyMsg = copyMsg.trim().split(' ');
+    copyMsg[0] = copyMsg[0].toLowerCase();
+    copyMsg = copyMsg.join(' ');
+    return copyMsg;
+  }
```
Comment (pret-a-porter) [inline lib/workers/repository/model/commit-message.ts:66]: Please, consider implementing new class

```
export class SemanticCommitMessage {
  readonly type: string;
  readonly scope?: string;
  readonly description: string;

  constructor(type: string, description: string, scope?: string) {
    this.type = type;
    this.description = description;
    this.scope = scope;
  }

  format(): string {
    // do something
  }
}
```

Comment (rarkins) [reply lib/workers/repository/model/commit-message.ts:66]: @RahulGautamSingh Do you mind if @pret-a-porter takes this one over?

Comment (RahulGautamSingh) [reply lib/workers/repository/model/commit-message.ts:66]: Nope, no worries.

**Pull request closed without merging**